tests/frontend/test_engine.py

```diff
@@ -170,6 +170,40 @@ def inspect():
         eng.run([p1, p2])
         assert inspect() == expected2
 
+    @pytest.mark.parametrize("eng", engines)
+    def test_combining_samples(self, eng):
+        """Check that samples are combined correctly when using multiple measurements"""
+
+        prog = sf.Program(5)
+        with prog.context as q:
+            ops.MeasureX | q[2]
+            ops.MeasureFock() | (q[1], q[3])
+            ops.MeasureFock() | q[0]
+
+        result = eng.run(prog)
+
+        # check that shape is (shots, measured_modes)
+        assert result.samples.shape == (1, 4)
+
+        # check that MesureFock measures `0` while MeasureX does NOT measure `0`.
+        correct_samples = [0, 0, 1, 0]
+        print(result.samples)
+        assert [bool(i) for i in result.samples[0]] == correct_samples
```

Comment:
Better to do it like this?
```suggestion
        assert all(i == j for i, j in zip(result.samples[0], correct_samples))
```

Reply:
Oh, this wouldn't actually do the same thing. It should only check that only some values (the 1 in `correct_samples` is not equal to 0). Another solution would be  `all(np.array(result.samples[0], dtype=bool) == correct_samples)` or `all(bool(i) == j for i, j in zip(result.samples[0], correct_samples))`. The original solution is actually fastest (1.33 µs) while your suggestion, with the `bool` casting, comes a close second (1.75 µs) and the `bool` casting via `np.array` takes a whooping 4.16 µs.